src/3rd_party_config.c

```diff
@@ -0,0 +1,296 @@
+	static int global_field_count = 0;
+	bool ret = false;
+
+	if (!strncmp(section, "REPO", strlen(section))) {
+		if (key && value) {
+			static struct repo *repo_var;
+			if (repo_field_count == 0) {
+				repo_var = (struct repo *)malloc(sizeof(struct repo));
+			}
+			if (!strncmp(key, "repo-name", strlen(key))) {
+				repo_var->repo_name = strdup_or_die(value);
+				repo_field_count++;
+				ret = true;
+			}
+
+			if (!strncmp(key, "repo-url", strlen(key))) {
+				repo_var->url = strdup_or_die(value);
+				repo_field_count++;
+				ret = true;
+			}
+
+			if (repo_field_count == MAX_REPO_ALLOWED_FIELDS) {
+				repos_kv = list_append_data(repos_kv, repo_var);
+				repo_field_count = 0;
+			}
+		}
+	} else if (!strncmp(section, "GLOBAL", strlen(section))) {
+		if (key && value) {
+			static struct global *global_var;
+			if (global_field_count > 2) {
+				return false;
+			}
+
+			if (!strncmp(key, "value1", strlen(key))) {
+				global_var->value1 = strdup_or_die(value);
+				global_field_count++;
+				ret = true;
+			}
+
+			if (!strncmp(key, "value2", strlen(key))) {
+				global_var->value2 = strdup_or_die(value);
+				global_field_count++;
+				ret = true;
+			}
+		}
+	}
+	return ret;
+}
+
+static int write_section(char *name)
+{
+	int ret = 0, fd;
+	ssize_t write_line_len;
+
+	fd = open(repo_config_file_path, O_WRONLY | O_APPEND);
+	if (fd < 0) {
+		return -1;
+	}
+
+	write_line_len = strlen(name);
+
+	if (write_line_len != write(fd, name, write_line_len)) {
+		error("error writing to file %d", errno);
+		ret = -1;
+	}
+
+	close(fd);
+	return ret;
+}
+
+static int write_kv(char *key, char *value)
+{
+	char *write_line;
+	int ret = 0;
+	ssize_t write_line_len;
+
+	int fd = open(repo_config_file_path, O_WRONLY | O_APPEND);
+	if (fd < 0) {
+		return -1;
+	}
+
+	string_or_die(&write_line, "%s=%s\n", key, value);
+	write_line_len = strlen(write_line);
+
+	if (write_line_len != write(fd, write_line, write_line_len)) {
+		error("error writing to file %d", errno);
+		ret = -1;
+	}
+
+	close(fd);
+	free_string(&write_line);
+	return ret;
+}
+
+static int write_empty_line(void)
+{
+	int ret = 0;
+	char *write_line = "\n";
+	ssize_t write_line_length = strlen(write_line);
+
+	int fd = open(repo_config_file_path, O_WRONLY | O_APPEND);
+	if (fd < 0) {
+		return -1;
+	}
+
+	if (write_line_length != write(fd,
+				       write_line,
+				       write_line_length)) {
+		error("error writing to file %d", errno);
+		ret = -1;
+	}
+
+	close(fd);
+	return ret;
+}
+
+static bool repo_config_read(void)
+{
+	return config_parse(repo_config_file_path, parse_key_values, NULL);
+}
+
+static void repo_config_read_free(void)
+{
+	list_free_list(repos_kv);
+}
+
+static void generate_path(void)
+{
+	char *repo_config_dir;
+	repo_config_dir = sys_path_join(globals.state_dir, REPO_CONFIG_DIR);
+	repo_config_file_path = sys_path_join(repo_config_dir, REPO_CONFIG_CONFIG_FILE);
+	free_string(&repo_config_dir);
+}
+
+static int ensure_repo_config_dir(void)
```

Comment:
the name of this function is misleading, "ensure" to me sounds more like a validation, to see if the config exists, while the function is really creating the directory. IMO it should be called something like "3rd_party_config_create_dir"

Reply:
Makes sense ..will change in next push